src/last_sign_state.rs

```diff
@@ -59,31 +53,34 @@ impl Display for LastSignError {
 
 impl LastSignState {
     pub fn load_state(path: &Path, chain_id: chain::Id) -> std::io::Result<LastSignState> {
-        if !path.exists() {
-            let mut lst = LastSignState {
-                data: EMPTY_DATA,
-                file: File::create(path)?,
-                _chain_id: chain_id,
-            };
-            lst.sync_to_disk()?;
-            return Ok(lst);
-        }
         let mut lst = LastSignState {
-            data: EMPTY_DATA,
-            file: OpenOptions::new().read(true).write(true).open(path)?,
+            data: LastSignData::default(),
+            path: path.to_owned(),
             _chain_id: chain_id,
         };
 
-        let mut contents = String::new();
-        lst.file.read_to_string(&mut contents)?;
-        lst.data = serde_json::from_str(&contents)?;
-        Ok(lst)
+        match fs::read_to_string(path) {
+            Ok(contents) => {
+                lst.data = serde_json::from_str(&contents)?;
+                Ok(lst)
+            }
+            Err(e) => {
+                if e.kind() == io::ErrorKind::NotFound {
+                    lst.sync_to_disk()?;
+                    Ok(lst)
+                } else {
+                    Err(e)
+                }
+            }
+        }
     }
 
     pub fn sync_to_disk(&mut self) -> std::io::Result<()> {
-        self.file
-            .write_all(serde_json::to_string(&self.data)?.as_ref())?;
-        self.file.sync_all()?;
```

Comment:
`atomicwrites` calls `fsync()` on the file's parent directory:

https://github.com/untitaker/rust-atomicwrites/blob/master/src/lib.rs#L166